### PR TITLE
Input group / Control group

### PR DESCRIPTION
**Pull request by Yozhgoor**
Making the component of the input group

Blueprint-doc: https://blueprintjs.com/docs/#core/components/text-inputs.input-group
Code ControlGroup: https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/forms/controlGroup.tsx
Code InputGroup: https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/forms/inputGroup.tsx

### CODE REVIEW

File: yewprint/src/control_group.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    fn view(&self) -> Html {
+        html! {
+            <div
+                class=(
+                    "bp3-control-group",
+                    self.props.fill.as_some("bp3-fill"),
+                    self.props.vertical.as_some("bp3-vertical"),
+                )
+            >
+            </div>
```
Comment (cecton) [inline yewprint/src/control_group.rs:46]: The only thing you need now is the children. You can take inspiration from [there](https://github.com/cecton/yewprint/blob/main/yewprint/src/buttons.rs#L71)

File: yewprint/src/input_group.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    fn view(&self) -> Html {
+        html! {
+            <div
+                class=(
+                    "bp3-input-group",
+                    self.props.disabled.as_some("bp3-disabled"),
+                    self.props.large.as_some("bp3-large"),
+                    self.props.small.as_some("bp3-small"),
+                )
```
Comment (cecton) [inline yewprint/src/input_group.rs:54]: You can compare to blueprint [here](https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/forms/inputGroup.tsx#L122-L128). You're still missing a few classes

File: yewprint/src/input_group.rs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    #[prop_or_default]
+    pub disabled: bool,
+    #[prop_or_default]
+    pub large: bool,
+    #[prop_or_default]
+    pub small: bool,
+    #[prop_or_default]
+    pub placeholder: String,
+    #[prop_or_default]
+    pub lefticon: Option<IconName>,
```
Comment (cecton) [inline yewprint/src/input_group.rs:20]: `left_icon`

File: yewprint/src/input_group.rs [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            >
+                {
+                    if let Some(icon) = self.props.lefticon {
+                        html! {
+                            <Icon icon=icon />
+                        }
+                    } else {
+                        html!()
+                    }
+                }
```
Comment (cecton) [inline yewprint/src/input_group.rs:64]: There is also a `left_element` which you can declare as `Option<VNode>`. See [here](https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/forms/inputGroup.tsx#L178-L180)

File: yewprint/src/input_group.rs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                {self.props.children.clone()}
+            </div>
```
Comment (cecton) [inline yewprint/src/input_group.rs:66]: Don't forget the right side: https://github.com/palantir/blueprint/blob/develop/packages/core/src/components/forms/inputGroup.tsx#L151

**Comment by Yozhgoor** [comment]
When i try to add another inputgroup i have this error
```
error: only one root html element is allowed (hint: you can wrap multiple html elements in a fragment `<></>`)
```
need some help for that and a review of the actual inputgroup to see how we can fix the difference between yewprint inputgroup and blueprint

**Comment by Yozhgoor** [comment]
i think it's not perfect but good enough for now
i need more component to make it looking great
what do you think about the actual work @cecton ?

File: yewprint-doc/src/input_group/mod.rs [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    fn create(_: Self::Properties, link: ComponentLink<Self>) -> Self {
+        InputGroupDoc {
+            callback: link.callback(|x| x),
+            state: ExampleProps {
+                disabled: false,
+                fill: false,
+                large: false,
+                small: false,
+                round: false,
+                placeholder: String::from(""),
```
Comment (cecton) [inline yewprint-doc/src/input_group/mod.rs:26]: Not used (so can be removed)

File: yewprint/src/buttons.rs [hunk 7]
```diff
@@ -22,6 +22,7 @@ pub struct Props {
     pub onclick: Callback<MouseEvent>,
     #[prop_or_default]
     pub class: String,
+    #[prop_or_default]
     pub children: html::Children,
```
Comment (cecton) [inline yewprint/src/buttons.rs:26]: Try with Option

**Review by cecton — CHANGES_REQUESTED** [review]
It's already nice. Close to be released!

**Comment by Yozhgoor** [comment]
you review is now reviewed, i still have an issue with the fill props of the control group

but all of your comment is fixed

File: yewprint-doc/src/control_group/example.rs [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+                fill=self.props.fill
+                vertical=self.props.vertical
+            >
+                <HtmlSelect<Option<Sorting>>
+                    options={vec![
+                        (None, "Filter".to_string()),
+                        (Some(Sorting::NameAscending), "Name-ascending".to_string()),
+                        (Some(Sorting::NameDescending), "Name-descending".to_string()),
+                        (Some(Sorting::PriceAscending), "Price-ascending".to_string()),
+                        (Some(Sorting::PriceDescending), "Price-descending".to_string()),
```
Comment (cecton) [inline yewprint-doc/src/control_group/example.rs:48]: Labels are still not the same than blueprint's doc

**Review by cecton — CHANGES_REQUESTED** [review]
ControlGroup: Vertical+Fill doesn't behave the same in the example on blueprint. Other than that it looks good

InputGroup: various work to do

**Comment by Yozhgoor** [comment]
For now i have some issues: 

inputGroup is visually bad, i don't know how to easily make a better render
i can't use the select for the last inputGroup, bad render with small and round props. Maybe use a menu like Blueprint
i can't use tag for now, bad render with small prop. Need to search to know what to put inside the tag

ControlGroup fill prop don't work

**Comment by Yozhgoor** [comment]
Need a review @cecton :D  
ControlGroup is fine i think
InputGroup need more enhancements but i think we can push it for now and come back on it later with some other component (like popover) not implemented yet

**Review by cecton — APPROVED** [review]
:ok_hand:

**Pull request merged**